modules/base_test.py

```diff
@@ -0,0 +1,69 @@
+"""
+Base class definitions to enable consistent logging. Requires ReFrame v. 3.11.x
+"""
+import math
+import reframe as rfm
+from reframe.core.decorators import run_after
+
+
+class DiRACTest(rfm.RegressionTest):
+
+    num_total_cores = variable(int, loggable=True)
+    num_omp_threads = variable(int, loggable=True)
+    num_mpi_tasks = variable(int, loggable=True)
+    num_mpi_tasks_per_node = variable(int, loggable=True)
+    num_nodes = variable(int, loggable=True)
```

Comment:
I think most if not all of these variables are standard ReFrame variables, and it'd be good to avoid duplication:

* `num_omp_threads` is [`num_cpus_per_task`](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.num_cpus_per_task)
* `num_mpi_tasks` is [`num_tasks`](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.num_tasks) (this is automatically used as argument to `mpiexec` to specify the number of MPI processes)
* `num_mpi_tasks_per_node` is [`num_tasks_per_node`](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.num_tasks_per_node)

`num_total_cores` and `num_nodes` are useful, but I think those can be inferred from other similar existing variables (which may still need to be added to the log)